pyresample/geometry.py

```diff
@@ -1372,22 +1377,26 @@ def proj4_string(self):
 
     def get_area_slices(self, area_to_cover):
         """Compute the slice to read based on an `area_to_cover`."""
-
         if not isinstance(area_to_cover, AreaDefinition):
             raise NotImplementedError('Only AreaDefinitions can be used')
 
         if self.proj_dict['proj'] != 'geos':
             raise NotImplementedError('Only geos supported')
 
         # Intersection only required for two different projections
-        if area_to_cover.proj_dict['proj'] == self.proj_dict['proj']:
+        if area_to_cover.proj_dict.get('proj') == self.proj_dict['proj']:
             logger.debug('Projections for data and slice areas are'
                          ' identical: %s', area_to_cover.proj_dict['proj'])
             # Get xy coordinates
             llx, lly, urx, ury = area_to_cover.area_extent
             x, y = self.get_xy_from_proj_coords([llx, urx], [lly, ury])
 
-            return slice(x[0], x[1] + 1), slice(y[1], y[0] + 1)
+            xstart = 0 if x[0] is np.ma.masked else x[0]
```

Comment:
Are 'old style' area definitions the only ones that return a masked value? What do dask arrays return?

Reply:
`self.get_xy_from_proj_coords` doesn't produce a dask array, but I suppose that would return np.nan ?

Reply:
Time to test I guess 😉

Reply:
Or is it not necessary since you are passing the extents which are scalars?

Reply:
yes, exactly